train_nsf_sim_cache_sid_load_pretrain.py

```diff
@@ -171,44 +154,45 @@ def run(rank, n_gpus, hps):
         net_g = DDP(net_g)
         net_d = DDP(net_d)
 
-    try:  # 如果能加载自动resume
-        _, _, _, epoch_str = utils.load_checkpoint(
-            utils.latest_checkpoint_path(hps.model_dir, "D_*.pth"), net_d, optim_d
-        )  # D多半加载没事
+    # try:  # 如果能加载自动resume
+    #     _, _, _, epoch_str = utils.load_checkpoint(
+    #         utils.latest_checkpoint_path(hps.model_dir, "D_*.pth"), net_d, optim_d
+    #     )  # D多半加载没事
+    #     if rank == 0:
+    #         logger.info("loaded D")
+    #     # _, _, _, epoch_str = utils.load_checkpoint(utils.latest_checkpoint_path(hps.model_dir, "G_*.pth"), net_g, optim_g,load_opt=0)
+    #     _, _, _, epoch_str = utils.load_checkpoint(
+    #         utils.latest_checkpoint_path(hps.model_dir, "G_*.pth"), net_g, optim_g
+    #     )
+    #     global_step = (epoch_str - 1) * len(train_loader)
+    #     # epoch_str = 1
+    #     # global_step = 0
+    # except:  # 如果首次不能加载，加载pretrain
+        
+    # traceback.print_exc()
+    epoch_str = 1
+    global_step = 0
+    if hps.pretrainG != "":
         if rank == 0:
-            logger.info("loaded D")
-        # _, _, _, epoch_str = utils.load_checkpoint(utils.latest_checkpoint_path(hps.model_dir, "G_*.pth"), net_g, optim_g,load_opt=0)
-        _, _, _, epoch_str = utils.load_checkpoint(
-            utils.latest_checkpoint_path(hps.model_dir, "G_*.pth"), net_g, optim_g
-        )
-        global_step = (epoch_str - 1) * len(train_loader)
-        # epoch_str = 1
-        # global_step = 0
-    except:  # 如果首次不能加载，加载pretrain
-        # traceback.print_exc()
-        epoch_str = 1
-        global_step = 0
-        if hps.pretrainG != "":
-            if rank == 0:
-                logger.info("loaded pretrained %s" % (hps.pretrainG))
-            print(
-                net_g.module.load_state_dict(
-                    torch.load(hps.pretrainG, map_location="cpu")["model"]
-                )
-            )  ##测试不加载优化器
-        if hps.pretrainD != "":
-            if rank == 0:
-                logger.info("loaded pretrained %s" % (hps.pretrainD))
-            print(
-                net_d.module.load_state_dict(
-                    torch.load(hps.pretrainD, map_location="cpu")["model"]
-                )
+            logger.info("loaded pretrained %s" % (hps.pretrainG))
+        print(
+            net_g.module.load_state_dict(
```

Comment:
why reinit the model on every call ?
can you make it a class and hold a "master" copy of the model loaded in __init__ ,
then copy and train on each call

Reply:
See previous comment.

Reply:
We have a different model here, depending on the target sample rate (which is a user defined parameter).
We can load all models (3 of them - `"f0G{sr}.pth"` with `sr` in `['32k', '40k', '48k']`), and the same for `"f0D{sr}.pth"`, if you think it's better (?).